roles/openshift_logging/tasks/set_defaults_from_current.yml

```diff
@@ -11,24 +11,236 @@
 ##  but fall back to using the value from a configmap as a default. If neither is set
 ##  then the variable remains undefined and the role default will be used.
 
-- conditional_set_fact:
+
+# Elasticsearch
+- when: openshift_logging_facts['elasticsearch']['configmaps']['logging-elasticsearch'] is defined
+  conditional_set_fact:
     facts: "{{ openshift_logging_facts['elasticsearch']['configmaps']['logging-elasticsearch']['elasticsearch.yml'] | flatten_dict }}"
     vars:
       __openshift_logging_es_number_of_shards: index.number_of_shards
       __openshift_logging_es_number_of_replicas: index.number_of_replicas
-  when: openshift_logging_facts['elasticsearch']['configmaps']['logging-elasticsearch'] is defined
 
-- conditional_set_fact:
+- when: openshift_logging_facts['elasticsearch']['deploymentconfigs'].keys() | count > 0
+  block:
+    - set_fact:
+        __es_dc_name: "{{ openshift_logging_facts['elasticsearch']['deploymentconfigs'].keys()[0] }}"
```

Comment:
The DC name is always the first entry?

Reply:
The keys for the deploymentconfig dictionary are the names of them 
https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_logging/library/openshift_logging_facts.py#L192